tests/stdlib/topenssl.nim

```diff
@@ -14,22 +14,18 @@ proc rsaPublicEncrypt*(fr: string): cstring =
   mKey.add "\n-----END PUBLIC KEY-----"
 
   var bio = bioNew(bioSMem())
-  if BIO_write(bio, mKey.cstring, mKey.len.cint) < 0:
-    raise newException(Exception, "ERROR")
+  doAssert not BIO_write(bio, mKey.cstring, mKey.len.cint) < 0

```

Comment:
always simplify: `doAssert BIO_write(bio, mKey.cstring, mKey.len.cint) >= 0`

but see https://linux.die.net/man/3/bio_write:

> All these functions return either the amount of data successfully read or written (if the return value is positive) or that no data was successfully read or written if the result is 0 or -1. If the return value is -2 then the operation is not implemented in the specific BIO type.

> A 0 or -1 return is not necessarily an indication of an error. In particular when the source/sink is non-blocking or of a certain type it may merely be an indication that no data is currently available and that the application should retry the operation later.